### ISSUE TITLE
Support an arbitrary number of clicks

### ISSUE THREAD

**Issue by IllyaMoskvin** [mathertel/OneButton#50]
It would be nice if this library could support triple clicks, quadruple clicks, etc.

Tentatively, it doesn't seem as if it'd be too hard to do this in a backwards-compatible way. I'd like to try to implement it. I recently made a project with a really nice wooden case, but only one button, so I'm pretty invested in seeing how much functionality can be squeezed out of a single button.

Would you be interested in accepting a pull request that solves this issue?

If so, in terms of approach, what are your thought about the following API?

```
void attachMultiClick(int clicks, callbackFunction newFunction);
```

I figure `attachClick` and `attachDoubleClick` can call `attachMultiClick` behind-the-scenes.

Probably related to the need behind #42.

**Comment by mathertel** [comment]
single and double clicks are the usual interactions we know from the web. long press and hold is there as well.
That's what the library was created for.
I lik to stay with this functionality because I personally don't like libraries that are huge just because they cover a lot of cases e.g. use them in a Tiny85 processor will be problematic. 
Extending as you did is exactly what should be done on a project basis.

**Comment by IllyaMoskvin** [comment]
Thanks for the reply, @mathertel. I understand where you're coming from with regard to wanting to keep the library lightweight. I do think that the `attachMultiClick` approach I described in the opening post is a generalization. Extending the current pattern to accommodate these use-cases does not feel like the best way to handle this. A refactor would be necessary. If implemented correctly, it may result in a smaller codebase. However, I think it may also require fundamental changes to the code that may result in it no longer being a finite state machine. Given this fact, I understand why you chose to close this issue. Supporting this functionality may not be in scope for this library.

For the project I've built, I ended up using @PatrickGlatz's fork. I had a need for triple-click, but no more. For the curious, my project's source code can be found here:

https://github.com/IllyaMoskvin/aic-nixie/blob/master/NixieMain/NixieMain.ino
